pkg/networking/middleware/response.go

```diff
@@ -0,0 +1,101 @@
+package middleware
+
+import (
+	"errors"
+	"net/http"
+
+	"github.com/snyk/error-catalog-golang-public/snyk"
+	"github.com/snyk/error-catalog-golang-public/snyk_errors"
+	"github.com/snyk/go-application-framework/pkg/configuration"
+	networktypes "github.com/snyk/go-application-framework/pkg/networking/network_types"
+)
+
+type ResponseMiddleware struct {
+	next       http.RoundTripper
+	config     configuration.Configuration
+	errHandler networktypes.ErrorHandlerFunc
+}
+
+func NewReponseMiddleware(roundTriper http.RoundTripper, config configuration.Configuration, errHandler networktypes.ErrorHandlerFunc) *ResponseMiddleware {
+	return &ResponseMiddleware{
+		next:       roundTriper,
+		config:     config,
+		errHandler: errHandler,
+	}
+}
+
+func (rm ResponseMiddleware) RoundTrip(req *http.Request) (*http.Response, error) {
+	res, err := rm.next.RoundTrip(req)
+
+	if err != nil {
+		return res, err
+	}
+
+	err = HandleResponse(res, rm.config)
+
+	if rm.errHandler != nil {
```

Comment:
Suggestion: To be more clear on what happens, I think we should remove the check if errHandler is nil, since this middleware is [only used](https://github.com/snyk/go-application-framework/pull/279/files#diff-99beec8232d7e64bc806ee6a0d9ee88bbff437a1a93094027686253ab80725a5R216) when an error handler is available.